src/any_agent/tools/a2a.py

```diff
@@ -85,3 +86,61 @@ async def _send_query(query: str) -> str:
             The result from the A2A agent, encoded in json.
     """
     return _send_query
+
+
+def _run_async(coro: Coroutine[Any, Any, Any]) -> Any:
+    """Run async code in a sync context.
+
+    This is useful because the tool is async, but we want to use it in a sync context.
+    """
+    try:
+        # Check if there's already an event loop running in this thread
+        asyncio.get_running_loop()
+
+        # If we get here, there IS a loop running, so we can't use asyncio.run()
+        # directly because it would try to create a second loop in the same thread
+        import concurrent.futures
+
+        def run_in_thread() -> Any:
+            # This runs in a fresh thread with no event loop, so asyncio.run() is safe
+            return asyncio.run(coro)
+
+        # Execute the async code in a separate thread to avoid loop conflicts
+        with concurrent.futures.ThreadPoolExecutor() as executor:
```

Comment:
Do you think it's worth it to define a package-level executor instead of creating a new one for each invocation?